src/lib/swapper/LifiSwapper/LifiSwapper.ts

```diff
@@ -167,18 +174,18 @@ export class LifiSwapper implements Swapper<EvmChainId> {
   /**
    * Get transactions related to a trade
    */
-  async getTradeTxs(tradeResult: TradeResult): Promise<TradeTxs> {
+  async getTradeTxs(tradeResult: TradeResult): Promise<Result<TradeTxs, SwapErrorRight>> {
     const getStatusRequest = this.executedTrades.get(tradeResult.tradeId)
 
     if (getStatusRequest === undefined) {
-      return { sellTxid: tradeResult.tradeId }
+      return Ok({ sellTxid: tradeResult.tradeId })
     }
 
     const statusResponse = await getLifi().getStatus(getStatusRequest)
 
-    return {
+    return Ok({
       sellTxid: tradeResult.tradeId,
       buyTxid: statusResponse.receiving?.txHash,
-    }
+    })
```

Comment:
id be curious to see what we can do here to provide polling updates - the lifi SDK actually provides human-readable messages about the status of the transaction as it get processed before the buyTxid arrives (i.e while the spinner is spinning). So we might be able to return something along the lines of

```
Promise<{
  sellTxid: Result<string, SomeHelpfulObject>
  buyTxid: Result<string, SomeHelpfulObject>
}>
```

Later of course

Reply:
Absolutely. Polling updates and displaying them to the user move us closer to a best-in-class swapper UX 🚀